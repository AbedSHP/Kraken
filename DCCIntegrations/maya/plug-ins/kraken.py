import sys, json

from PySide import QtGui

import kraken.ui
from kraken.ui.kraken_ui import KrakenUI

from maya import cmds
# import maya.OpenMaya as OpenMaya
import maya.OpenMayaUI as OpenMayaUI
import maya.OpenMayaMPx as OpenMayaMPx

try:
  # Maya 2013 with custom pyside build
  import PySide.shiboken as shiboken
except:
  # Maya 2014 and higher
  import shiboken



# Command
class OpenKrakenEditorCommand(OpenMayaMPx.MPxCommand):
  def __init__(self):
    OpenMayaMPx.MPxCommand.__init__(self)

  # Invoked when the command is run.
  def doIt(self,argList):
    parent = OpenMayaUI.MQtUtil.mainWindow()
    parent = shiboken.wrapInstance(long(parent), QtGui.QWidget)
    dfgApp = KrakenUI(parent)

  # Creator
  @staticmethod
  def creator():
    return OpenMayaMPx.asMPxPtr( OpenKrakenEditorCommand() )


class KrakenUndoableCmd(OpenMayaMPx.MPxCommand):

  def __init__(self):
    OpenMayaMPx.MPxCommand.__init__(self)

  def isUndoable(self):
    return True

  def doIt(self, argList):
    return 0

  def redoIt(self):
    SingletonHost.getController().redo()
    return 0

  def undoIt(self):
    SingletonHost.getController().undo()
    return 0

  @staticmethod
  def creator():
    return OpenMayaMPx.asMPxPtr( KrakenUndoableCmd() )


# Initialize the script plug-in
def initializePlugin(mobject):
  mplugin = OpenMayaMPx.MFnPlugin(mobject)

  try:
    mplugin.registerCommand( 'openKrakenEditor', OpenKrakenEditorCommand.creator )
  except:
    sys.stderr.write( 'Failed to register DFG commands:openKrakenEditor' )
    raise

  try:
    mplugin.registerCommand( 'krakenUndoableCmd', KrakenUndoableCmd.creator )
  except:
    sys.stderr.write( 'Failed to register DFG commands:krakenUndoableCmd' )
    raise

# Uninitialize the script plug-in
def uninitializePlugin(mobject):
  mplugin = OpenMayaMPx.MFnPlugin(mobject)

  try:
    mplugin.deregisterCommand( 'openKrakenEditor' )
  except:
    sys.stderr.write( 'Failed to unregister command: openKrakenEditor' )


  try:
    mplugin.deregisterCommand( 'krakenUndoableCmd' )
  except:
    sys.stderr.write( 'Failed to unregister command: krakenUndoableCmd' )
