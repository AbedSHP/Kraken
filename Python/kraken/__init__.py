"""Kraken Framework."""

__all__ = ['core', 'examples', 'helpers', 'plugins', 'gui']