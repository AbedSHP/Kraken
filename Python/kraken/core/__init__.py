"""Kraken Core."""

VERSION_MAJOR = 0
VERSION_MINOR = 0
VERSION_BUILD = 1
VERSION_SUFFIX = "alpha"


def getVersion():
    """Contatenates the version globals and returns the current version of
    Kraken.

    Return:
    String, current version of Kraken.

    """

    versionString = str(VERSION_MAJOR) + "." + str(VERSION_MINOR) + "." + str(VERSION_BUILD)
    if VERSION_SUFFIX:
        versionString = versionString + "-" + VERSION_SUFFIX

    return versionString
