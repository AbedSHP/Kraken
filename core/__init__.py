"""Kraken Framework."""

VERSION_MAJOR = 0
VERSION_MINOR = 0
VERSION_BUILD = 1
VERSION_SUFFIX = "beta"


def getVersion():
    versionString = str(VERSION_MAJOR) + "." + str(VERSION_MINOR) + "." + str(VERSION_BUILD)
    if VERSION_SUFFIX:
        versionString = versionString + "-" + VERSION_SUFFIX

    return versionString