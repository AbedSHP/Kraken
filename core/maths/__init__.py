"""Kraken - math module."""


from vec2 import Vec2
from vec3 import Vec3
from vec4 import Vec4
from quat import Quat
from euler import Euler
from xfo import Xfo


PI = 3.141592653589793
DEG_TO_RAD = 0.017453292519943295
RAD_TO_DEG = 57.29577951308232


def Math_radToDeg(val):
    """Converts radians to degrees.

    Arguments:
    val -- Scalar, value to convert to degrees.

    Return:
    Scalar, value in degrees.

    """

    return val * RAD_TO_DEG


def Math_degToRad(val):
    """Converts degrees to radians.

    Arguments:
    val -- Scalar, value to convert to radians.

    Return:
    Scalar, value in radians.

    """

    return val * DEG_TO_RAD