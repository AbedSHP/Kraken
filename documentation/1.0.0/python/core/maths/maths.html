

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Maths &mdash; Kraken Rigging Framework 1.0.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../../../_static/css/theme.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../../../_static/lowdown.css" type="text/css" />
  

  
    <link rel="top" title="Kraken Rigging Framework 1.0.0 documentation" href="../../../index.html"/>
        <link rel="up" title="Kraken Core" href="../core.html"/>
        <link rel="next" title="Euler" href="euler.html"/>
        <link rel="prev" title="Kraken Saver" href="../io/kraken_saver.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        

        
          <a href="../../../index.html" class="icon icon-home"> Kraken Rigging Framework
        

        
        </a>

        
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

        
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
        
          
          
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../../../license.html">License</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../installation.html">Installation</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../../releases.html">Releases</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../../../releases/release_1.0.0.html">Release 1.0.0</a><ul class="simple">
</ul>
</li>
</ul>
</li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../integrations.html">Integrations</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../../../integrations/maya.html">Maya Integration</a><ul class="simple">
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../integrations/softimage.html">Softimage Integration</a><ul class="simple">
</ul>
</li>
</ul>
</li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../user_interface.html">User Interface</a><ul class="simple">
</ul>
</li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../kl/kl_guide.html">KL Guide</a><ul>
<li class="toctree-l2"><a class="reference internal" href="../../../kl/kl_library.html">KL Library</a><ul>
<li class="toctree-l3"><a class="reference internal" href="../../../kl/solvers/kraken_solver.html">Kraken Solver</a><ul class="simple">
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../../kl/solvers/kraken_arg.html">Kraken Arg</a><ul class="simple">
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../../kl/solvers/bezier_spine_solver.html">Bezier Spine Solver</a><ul class="simple">
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../../kl/solvers/kraken_multiposecon_solver.html">Kraken Multi-Pose Constraint Solver</a><ul class="simple">
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../../kl/solvers/kraken_posecon_solver.html">Kraken Pose Constraint Solver</a><ul class="simple">
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../../kl/solvers/kraken_rigscale_solver.html">Kraken Rig Scale Solver</a><ul class="simple">
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../../kl/solvers/nbone_ik_solver.html">N-Bone IK Solver</a><ul class="simple">
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../../kl/solvers/twobone_ik_solver.html">Two-Bone IK Solver</a><ul class="simple">
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../../kl/tutorials/custom_solvers.html">Custom Solvers</a><ul class="simple">
</ul>
</li>
</ul>
</li>
</ul>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../../python_guide.html">Python Guide</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../../python_library.html">Python Library</a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="../core.html">Kraken Core</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="../builder.html">Builder</a></li>
<li class="toctree-l4"><a class="reference internal" href="../kraken_system.html">Kraken System</a></li>
<li class="toctree-l4"><a class="reference internal" href="../profiler.html">Profiler</a></li>
<li class="toctree-l4"><a class="reference internal" href="../synchronizer.html">Synchronizer</a></li>
<li class="toctree-l4"><a class="reference internal" href="../configs/config.html">Configs</a></li>
<li class="toctree-l4"><a class="reference internal" href="../io/io.html">IO</a></li>
<li class="toctree-l4 current"><a class="current reference internal" href="">Maths</a></li>
<li class="toctree-l4"><a class="reference internal" href="../objects/objects.html">Objects</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../helpers/helpers.html">Helpers</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../../helpers/utility_methods.html">Utility Methods</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../plugins/plugins.html">Plugins</a><ul>
<li class="toctree-l4"><a class="reference internal" href="../../plugins/maya_plugin/maya_plugin.html">Maya Plugin</a></li>
<li class="toctree-l4"><a class="reference internal" href="../../plugins/si_plugin/si_plugin.html">Softimage Plugin</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../../ui/ui.html">Kraken UI</a><ul class="simple">
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../tutorials/custom_components.html">Custom Components</a><ul class="simple">
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../tutorials/scripted_builds.html">Scripted Builds</a><ul class="simple">
</ul>
</li>
</ul>
</li>
</ul>

          
        
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../../../index.html">Kraken Rigging Framework</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../../../index.html">Docs</a> &raquo;</li>
      
          <li><a href="../../python_guide.html">Python Guide</a> &raquo;</li>
      
          <li><a href="../../python_library.html">Python Library</a> &raquo;</li>
      
          <li><a href="../core.html">Kraken Core</a> &raquo;</li>
      
    <li>Maths</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../../../_sources/python/core/maths/maths.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document">
            
  <div class="section" id="maths">
<h1>Maths<a class="headerlink" href="#maths" title="Permalink to this headline">¶</a></h1>
<p>This is the math library for Kraken. We wrap RTVal calls from KL to Python objects and expose the methods to keep our math library in line with the KL math types.</p>
<div class="section" id="modules">
<h2>Modules<a class="headerlink" href="#modules" title="Permalink to this headline">¶</a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="euler.html">Euler</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="mat33.html">Mat33</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="mat44.html">Mat44</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="math_object.html">Math Object</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="quat.html">Quat</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="rotation_order.html">Rotation Order</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="vec2.html">Vec2</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="vec3.html">Vec3</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="vec4.html">Vec4</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="xfo.html">Xfo</a><ul class="simple">
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="module-kraken.core.maths">
<span id="module-methods"></span><h2>Module Methods<a class="headerlink" href="#module-kraken.core.maths" title="Permalink to this headline">¶</a></h2>
<p>Kraken - math module.</p>
<dl class="function">
<dt id="kraken.core.maths.Math_radToDeg">
<code class="descname">Math_radToDeg</code><span class="sig-paren">(</span><em>val</em><span class="sig-paren">)</span><a class="headerlink" href="#kraken.core.maths.Math_radToDeg" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts radians to degrees.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>val</strong> (<em>float</em>) &#8211; value to convert to degrees.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Value in degrees.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="kraken.core.maths.Math_degToRad">
<code class="descname">Math_degToRad</code><span class="sig-paren">(</span><em>val</em><span class="sig-paren">)</span><a class="headerlink" href="#kraken.core.maths.Math_degToRad" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts degrees to radians.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>val</strong> (<em>float</em>) &#8211; value to convert to radians.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Value in radians.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="kraken.core.maths.decodeValue">
<code class="descname">decodeValue</code><span class="sig-paren">(</span><em>jsonData</em><span class="sig-paren">)</span><a class="headerlink" href="#kraken.core.maths.decodeValue" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a constructed math value based on the provided json data.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>jsondata</strong> (<em>dict</em>) &#8211; The JSON data to use to decode into a Math value.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">The constructed math value</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">object</td>
</tr>
</tbody>
</table>
</dd></dl>

<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
</div>
<div class="section" id="indices-and-tables">
<h2>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><a class="reference internal" href="../../../genindex.html"><span>Index</span></a></li>
<li><a class="reference internal" href="../../../py-modindex.html"><span>Module Index</span></a></li>
<li><a class="reference internal" href="../../../search.html"><span>Search Page</span></a></li>
</ul>
<div class="line-block">
<div class="line"><br /></div>
</div>
</div>
</div>


          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="euler.html" class="btn btn-neutral float-right" title="Euler" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../io/kraken_saver.html" class="btn btn-neutral" title="Kraken Saver" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2015, Eric Thivierge, Phil Taylor, Hybride Technologies, Fabric Software Inc..
    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../../../',
            VERSION:'1.0.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../../../_static/jquery.js"></script>
      <script type="text/javascript" src="../../../_static/underscore.js"></script>
      <script type="text/javascript" src="../../../_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="../../../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>