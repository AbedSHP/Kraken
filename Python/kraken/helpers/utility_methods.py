

def logHierarchy(kObject):
    """Traverses the given Kraken hierarchy and logs the names of all the objects.

    Return:
    None

    """

    print kObject.getFullName()
    for i in xrange(kObject.getNumChildren()):
        child = kObject.getChildByIndex(i)
        logHierarchy(child)



from kraken.core.maths.math_object import MathObject
from kraken.core.maths import decodeValue

def __convertFromJSON(jsonData):

    if type(jsonData) is list:
        newList = []
        for item in jsonData:
            newList.append(__convertFromJSON(item))
        return newList
    elif type(jsonData) is dict:
        if '__mathObjectClass__' in jsonData.keys():
            return decodeValue(jsonData)
        for key, value in jsonData.iteritems():
            jsonData[key] = __convertFromJSON(value)
    return jsonData

def prepareToLoad(jsonData):

    return __convertFromJSON(jsonData)

def __convertToJSON(jsonData):

    if isinstance(jsonData, MathObject):
        return jsonData.jsonEncode()
    elif type(jsonData) is list:
        newList = []
        for item in jsonData:
            newList.append(__convertToJSON(item))
        return newList
    elif type(jsonData) is dict:
        for key, value in jsonData.iteritems():
            jsonData[key] = __convertToJSON(value)
    return jsonData

def prepareToSave(jsonData):
    return __convertToJSON(jsonData)