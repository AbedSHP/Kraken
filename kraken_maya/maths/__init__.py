from kraken.core.maths import *