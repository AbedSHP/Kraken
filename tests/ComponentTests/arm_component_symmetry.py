import json

from kraken import plugins
from kraken.core.maths import Xfo, Vec3, Quat

from kraken_examples.arm_component import ArmComponentGuide, ArmComponentRig

from kraken.helpers.utility_methods import logHierarchy

leftArmGuide = ArmComponentGuide('arm')
leftArmGuide.loadData({
        'name': 'Arm',
        'location': 'L',
        'bicepXfo': Xfo(Vec3(2.27, 15.295, -0.753)),
        'forearmXfo': Xfo(Vec3(5.039, 13.56, -0.859)),
        'wristXfo': Xfo(Vec3(7.1886, 12.2819, 0.4906)),
        'handXfo': Xfo(tr=Vec3(7.1886, 12.2819, 0.4906),
                       ori=Quat(Vec3(-0.0865, -0.2301, -0.2623), 0.9331)),
        'bicepFKCtrlSize': 1.75,
        'forearmFKCtrlSize': 1.5
    })

# Save the arm guid data for persistence.
rightArmGuide = ArmComponentGuide('arm')
rightArmGuide.setLocation('R')

rightArmGuide.pasteData(leftArmGuide.copyData())

builder = plugins.getBuilder()
builder.build(leftArmGuide)
builder.build(rightArmGuide)


# armRigData = armGuide.getRigBuildData()

# armLeft = ArmComponentRig()
# armLeft.loadData(armGuide.getRigBuildData())

# armRight = ArmComponentRig()
# armRight.loadData(armGuide.getRigBuildData())

# print '==armLeft=='
# logHierarchy(armLeft)

