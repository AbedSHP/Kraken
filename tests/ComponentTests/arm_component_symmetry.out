Failed to find DCC builder. Falling back to Python builder.
==armLeft==
Arm:L
Arm:L.controls
Arm:L.controls.Arm:L
Arm:L.controls.Arm:L.inputs
Arm:L.controls.Arm:L.inputs.globalSRT
Arm:L.controls.Arm:L.inputs.clavicleEnd
Arm:L.controls.Arm:L.outputs
Arm:L.controls.Arm:L.outputs.bicep
Arm:L.controls.Arm:L.outputs.forearm
Arm:L.controls.Arm:L.outputs.armEndXfo
Arm:L.controls.Arm:L.outputs.hand
Arm:L.controls.Arm:L.bicepFK
Arm:L.controls.Arm:L.bicepFK.bicepFK
Arm:L.controls.Arm:L.bicepFK.bicepFK.forearmFK
Arm:L.controls.Arm:L.bicepFK.bicepFK.forearmFK.forearmFK
Arm:L.controls.Arm:L.hand
Arm:L.controls.Arm:L.hand.hand
Arm:L.controls.Arm:L.IK
Arm:L.controls.Arm:L.IK.IK
Arm:L.controls.Arm:L.UpV
Arm:L.controls.Arm:L.UpV.UpV
Arm:L.deformers
Arm:L.deformers.Arm:L
Arm:L.deformers.Arm:L.bicep
Arm:L.deformers.Arm:L.forearm
Arm:L.deformers.Arm:L.wrist
Arm:L.deformers.Arm:L.hand
==armRight==
Arm:R
Arm:R.controls
Arm:R.controls.Arm:R
Arm:R.controls.Arm:R.inputs
Arm:R.controls.Arm:R.inputs.globalSRT
Arm:R.controls.Arm:R.inputs.clavicleEnd
Arm:R.controls.Arm:R.outputs
Arm:R.controls.Arm:R.outputs.bicep
Arm:R.controls.Arm:R.outputs.forearm
Arm:R.controls.Arm:R.outputs.armEndXfo
Arm:R.controls.Arm:R.outputs.hand
Arm:R.controls.Arm:R.bicepFK
Arm:R.controls.Arm:R.bicepFK.bicepFK
Arm:R.controls.Arm:R.bicepFK.bicepFK.forearmFK
Arm:R.controls.Arm:R.bicepFK.bicepFK.forearmFK.forearmFK
Arm:R.controls.Arm:R.hand
Arm:R.controls.Arm:R.hand.hand
Arm:R.controls.Arm:R.IK
Arm:R.controls.Arm:R.IK.IK
Arm:R.controls.Arm:R.UpV
Arm:R.controls.Arm:R.UpV.UpV
Arm:R.deformers
Arm:R.deformers.Arm:R
Arm:R.deformers.Arm:R.bicep
Arm:R.deformers.Arm:R.forearm
Arm:R.deformers.Arm:R.wrist
Arm:R.deformers.Arm:R.hand
